cloudinit/sources/DataSourceEc2.py

```diff
@@ -731,28 +735,100 @@ def convert_ec2_metadata_network_config(network_md, macs_to_nics=None,
        not provided, get_interfaces_by_mac is called to get it from the OS.
     @param: fallback_nic: Optionally provide the primary nic interface name.
        This nic will be guaranteed to minimally have a dhcp4 configuration.
+    @param: apply_network_config: Boolean set True to configure all networking
+       presented by IMDS. This includes rendering secondary IPv4 and IPv6
+       addresses on all NICs and rendering network config on secondary NICs.
+       If False, only the primary nic will be configured and only with dhcp
+       (IPv4/IPv6).
 
-    @return A dict of network config version 1 based on the metadata and macs.
+    @return A dict of network config version 2 based on the metadata and macs.
     """
-    netcfg = {'version': 1, 'config': []}
+    netcfg = {'version': 2, 'ethernets': {}}
     if not macs_to_nics:
         macs_to_nics = net.get_interfaces_by_mac()
     macs_metadata = network_md['interfaces']['macs']
-    for mac, nic_name in macs_to_nics.items():
+
+    if not apply_network_config:
+        LOG.debug(
+            'Skipping network configuration for secondary NICs and IPs.'
+            'Datasource apply_network_config value is set False')
```

Comment:
Let's drop this;  I don't think we logged this before and if the number of nics is 1, we emit this log for no reason.

Reply:
ahh missed this. Fixed now.
0b6fa316624f205f0db47d6a3087d99ab3acf6f6